sys/include/net/ng_netreg.h

```diff
@@ -126,6 +126,30 @@ int ng_netreg_num(ng_nettype_t type, uint32_t demux_ctx);
  */
 ng_netreg_entry_t *ng_netreg_getnext(ng_netreg_entry_t *entry);
 
+/**
+ * @brief   Calculates the checksum for a packet of type @p checksum_hdr_type.
+ *
+ * @param[in] pseudo_hdr_type   The type of the pseudo header if required.
+ *                              May be NG_NETTYPE_UNDEF if no pseudo header is
+ *                              required for the checksum calculation.
+ * @param[in] csum_hdr_type     The type of the packet the checksum should be
+ *                              calculated for
+ * @param[in,out] pkt           The packet the checksum should be calculated
+ *                              for.
+ *
+ * @return  0, on success.
+ * @return  -EBADMSG, @p pkt did not contain a header of type @p pseudo_hdr_type
+ *          (if not NG_NETTYPE_UNDEF) or @p csum_hdr_type
+ * @return  -EFAULT, @p pkt was NULL.
+ * @return  -EINVAL, if @p pseudo_hdr_type is NG_NETTYPE_UNDEF but a pseudo
+ *          header was required
+ * @return  -ENOENT, if @ref net_netreg does not know how to calculate checksum
+ *          for @p csum_hdr_type
+ */
+int ng_netreg_calculate_csum(ng_nettype_t pseudo_hdr_type,
+                             ng_nettype_t csum_hdr_type,
```

Comment:
and for the `pseudo_hdr_type`: thats mainly because there might be protocols in the future (IPv6 encapsulated IPv4 e.g.) where `pkt->type` might not the `pseudo_hdr_type` either.